spec/models/activity_spec.rb

```diff
@@ -140,16 +140,20 @@
       let(:product) { activity.price_list.products.sample }
       let(:product_price) { activity.price_list.product_price_for(product).price }
       let(:cash_order) { FactoryBot.create(:order, :cash, activity: activity) }
+      let(:pin_order) { FactoryBot.create(:order, :pin, activity: activity) }
       let(:order) { FactoryBot.create(:order, activity: activity) }
 
       before do
         FactoryBot.create(:order_row, product: product, order: cash_order, product_count: 2)
+        FactoryBot.create(:order_row, product: product, order: pin_order, product_count: 2)
```

Comment:
Please make 4 pin orders, otherwise the cash orders and pin orders can be inversed without this test failing